src/main/java/mat/server/CQLServiceImpl.java

```diff
@@ -102,18 +103,21 @@
 
 	private static final int COMMENTS_MAX_LENGTH = 2500;
 
-	@Autowired 
+	@Autowired
 	private CQLLibraryDAO cqlLibraryDAO;
-	
-	@Autowired 
+
+	@Autowired
 	private CQLLibraryAssociationDAO cqlLibraryAssociationDAO;
-	
-	@Autowired 
+
+	@Autowired
 	private MeasurePackageService measurePackageService;
-	
+
 	@Autowired
 	private UserDAO userDAO;
 
+	@Autowired
+	private FeatureFlagServiceImpl featureFlagService;

```

Comment:
why do you inject am Impl and not a corresponding interface?

Reply:
this is because the @Service is on impl and not on interface i guess. interface did not work

Reply:
that should work in any case on interface. regardless of where the annotation is placed.
if it's not, then something is broken in our spring jar dependencies